### ISSUE TITLE
Why stopPropagation when click on Dropdown element.

### ISSUE THREAD

**Issue by hungdoan2** [twbs/bootstrap#17008]
I wrote a short demo in this link: 
[Demo](http://plnkr.co/edit/CvQ0XsggIgXQ8acz1nSm)
 

The important part is : 
```
 <script>
      $(document).on("click", function(){
        alert("Document click");
      });
    </script>
```
The problem is that `bootstrap dropdown`  prevent me to listen for `click event` on `document`.
**So what is the reason to stopPropagation when click on a `Dropdown` ?**

p/s: I try to resolve this problem, It work fine with me now. But I'm not sure if my fix effect to others function/features 
```
  Dropdown.prototype.toggle = function (e) {
    ...
    other code 
    ...
    //return false  <- remove return false (which causes stopPropagation)
  }
```
And 
```
  function clearMenus(e) {
    //Because of I do not  stopPropagation
    //THEN $(document).on('click.bs.dropdown.data-api', clearMenus) will be trigged!
    //THEN the dropdown will be closed immediately
    // So, to make sure It doesn't closed after I open it. I won't close the menu(dropdown) if the source event come from the toggle element
    if (e && $(e.target).closest(toggle).length > 0) {
        return;
    }
    ...
    other code
    ...
  }
```

**Comment by twbs-lmvtfy** [comment]
Hi @hungdoan2!

You appear to have posted a live example (http://run.plnkr.co/plunks/CvQ0XsggIgXQ8acz1nSm/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W001](https://github.com/twbs/bootlint/wiki/W001): `<head>` is missing UTF-8 charset `<meta>` tag
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by kkirsche** [comment]
If I remember correctly, I believe it's due to file input fields within drop down menus to prevent the dropdown from closing when the click event is launched to browser for the file.

**Comment by hungdoan2** [comment]
@kkirsche Actually I don't think so. Because the  `Dropdown.prototype.toggle` is executed only when you click on `[data-toggle="dropdown"]`, I don't see any relation to file input.
A relative problem that I can image is `mixing dropdown menu` like [this](http://plnkr.co/edit/XxcxgexDWwVSkzOWXlhT?p=preview) ,but I don't think this is an use case of bootstrap.

Back to the issue, 
I have reviewed carefully again.
1. The problem is caused in  `Dropdown.prototype.toggle`
2. `Dropdown.prototype.toggle` is executed only when click on `[data-toggle="dropdown"]`
3. The `clearMenus(e)` is executed when click on the `document` or when `Dropdown.prototype.toggle`
In my point of view, If It's backwards-compatible and doesn't make a change request. (Actually I have not run test suite for It, yet - I will setup and test it later)
**Any way. I think It would be great if next version of bootstrap can resolve this. Because of stopPropagation can lead to some `confusing issue` when working with other library.**

**Comment by selvagsz** [comment]
Bitten by this issue. Is there any updates on this ?

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team
